src/components/webview/CodemodListProvider.ts

```diff
@@ -133,23 +137,45 @@ export class CodemodListPanelProvider implements WebviewViewProvider {
 		}
 
 		if (message.kind === 'webview.global.afterWebviewMounted') {
-			this.getCodemodTree();
+			this.getCodemodTree('recommended');
 		}
 	};
 
-	public async getCodemodTree() {
-		await this.__codemodService.getCodemods();
-		const codemodList = this.__getCodemod();
-		const treeNodes = codemodList.map((codemod) =>
-			this.__getTreeNode(codemod),
-		);
+	public async getCodemodTree(type: 'recommended' | 'public') {
+		const recommended = type === 'recommended';
+		try {
+			if (recommended) {
+				await this.__codemodService.getCodemods();
+			} else {
+				await this.__codemodService.getDiscoverdCodemods();
+			}
+			const codemodList = this.__getCodemod(recommended);
+			const treeNodes = codemodList.map((codemod) =>
+				this.__getTreeNode(codemod),
+			);
```

Comment:
let's validate treeNodes here

Reply:
Validate `treeNodes` here so you don't need to do `treeNodes?.[0]` below

Reply:
the first tree node might not exist, but the view has to be created.